index.d.ts

```diff
@@ -13,11 +13,11 @@ type NoStreamStdioOption =
 	| Writable
 	| [NoStreamStdioOption];
 
-type BaseStdioOption =
+type BaseStdioOption<IsSync extends boolean = boolean> =
 	| 'pipe'
-	| 'overlapped'
 	| 'ignore'
-	| 'inherit';
+	| 'inherit'
+	| IfAsync<IsSync, 'overlapped'>;
```

Comment:
Note: `overlapped` currently crashes the process with `spawnSync()`, instead of just throwing an error, or just behaving like `pipe`.

I opened a bug in https://github.com/nodejs/node/issues/52265